pkg/syncer/upsync/upsync_reconcile.go

```diff
@@ -0,0 +1,261 @@
+	}
+	resourceVersionDownstream := downstreamResource.GetResourceVersion()
+	markedForDeletionDownstream := downstreamResource.GetDeletionTimestamp() != nil
+
+	if upstreamObject == nil && !markedForDeletionDownstream {
+		// Resource doesn't exist upstream => let's create it upstream unless the corresponding downstream resource has been marked for deletion.
+		logger.Info("Creating resource upstream")
+		preparedResource := c.prepareResourceForUpstream(ctx, gvr, upstreamNamespace, upstreamClusterName, downstreamResource)
+
+		if !dirtyStatus {
+			// if no status needs to be upsynced upstream, then we can set the resource version annotation at the same time as we create the
+			// resource
+			preparedResource.SetAnnotations(addResourceVersionAnnotation(resourceVersionDownstream, preparedResource.GetAnnotations()))
+			// Create the resource
+			_, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Create(ctx, preparedResource, metav1.CreateOptions{})
+			return false, err
+		}
+
+		// Status also needs to be upsynced so let's do it in 3 steps:
+		// - create the resource
+		createdResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Create(ctx, preparedResource, metav1.CreateOptions{})
+		if err != nil {
+			return false, err
+		}
+		// - update the status as a distinct action,
+		preparedResource.SetResourceVersion(createdResource.GetResourceVersion())
+		updatedResource, err := c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).UpdateStatus(ctx, preparedResource, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+		// - finally update the main content again to set the resource version annotation to the value of the downstream resource version
+		preparedResource.SetAnnotations(addResourceVersionAnnotation(resourceVersionDownstream, preparedResource.GetAnnotations()))
+		preparedResource.SetResourceVersion(updatedResource.GetResourceVersion())
+		_, err = c.upstreamClient.Resource(gvr).Cluster(upstreamClusterName.Path()).Namespace(upstreamNamespace).Update(ctx, preparedResource, metav1.UpdateOptions{})
+		return false, err
+	}
+
+	resourceVersionUpstream := upstreamObject.GetAnnotations()[ResourceVersionAnnotation]
+	if resourceVersionDownstream != resourceVersionUpstream {
+		// Resource version annotation on the upstream resource doesn't match the downstream resource version
+		// => let's update the resource upstream.
+		logger.Info("Updating upstream resource")
+		preparedResource := c.prepareResourceForUpstream(ctx, gvr, upstreamNamespace, upstreamClusterName, downstreamResource)
+		if err != nil {
+			return false, err
+		}
+
+		if !dirtyStatus {
+			// if no status needs to be upsynced upstream, then we can set the resource version annotation to the downstream resource version
+			// at the same time as we update the main resource content.
```

Comment:
```suggestion
		// quick path: status unchanged, only update main resource
		if !dirtyStatus {
```

Reply:
Fixed in commit fffff6927c99c86702c7709881a77057331950cd